### ISSUE TITLE
[Suggestion] Use JCF-CLI as code formatter.

### ISSUE THREAD

**Issue by pakLebah** [alefragnani/vscode-language-pascal#31]
Hi,

I've just made a CLI version of JCF (Jedi Code Formatter) which is a code formatter widely used in Pascal/Delphi world. Unlike the original JCF, and as it's a Free Pascal project, it need to be compiled using FreePascal through Lazarus IDE or VS Code, so it works cross platform natively.

I have used it myself in VS Code successfully through task. See the demo in the project's readme. Perhaps you want to use it with your extension. You may take a look at the JCF-CLI project [here](http://github.com/git-bee/jcf-cli).

HTH.

**Comment by pakLebah** [comment]
@alefragnani: Yes, as JCF-CLI is just JCF without GUI-lib dependency, it behaves exactly like the original JCF. JCF only works on entire file which should also be well formed. I personally don't think it's a limitation, because I usually use a code formatter after I finish with the code, not during the writing of the code. But that's me, ymmv of course. 😊

**Comment by alefragnani** [comment]
Well, I prefer to have _options_. Sometimes I paste a code from somewhere and it's not formatted as I like, so I have to format that piece of code. I don't have good experiences with _post formatting_, so I prefer _format when I want_ :)

About the `jcf-cli.exe` itself. If the command line options does not differ from `jcf.exe`, there is no need to _adapt_ the extension to use it, but simply update the `README.md` to indicate it's use.

**Comment by pakLebah** [comment]
JCF-CLI is JCF _without_ GUI-lib dependency. It means:

1. The code can be compiled using _only_ FPC, no need to install Lazarus IDE.
2. The executable is a single file and can be run from a terminal console.
3. It's truly cross platform (any platforms supported by FPC).

That's not possible using the original JCF. I sometimes write Pascal code on Linux server with no GUI library installed. I made JCF-CLI solely for that purpose. And I think, JCF-CLI is also appropriate to be used by VS Code extension like yours, instead of the original JCF. At least to provide an option to JCF users on non-Windows platforms.

**Comment by alefragnani** [comment]
I did understand its concept 😄 . 

I'm just asking if I could use the same _command line parameters_ as `jcf.exe` to format files. If yes, there is no need to _change the extension to support it_ but simply update the `README` to say that it can use `jcf.exe` _or_ `jcf-cli.exe`.

Did you try to use it in `pascal.formatter.enginePath` setting?

**Comment by pakLebah** [comment]
@alefragnani:
> Did you try to use it in `pascal.formatter.enginePath` setting?

Sorry, I don't use your extension. I prefer OmniPascal since it provides better code navigation and outline. I use code formatter through task anyway.

**Comment by alefragnani** [comment]
Well, that's weird 🤔
